src/nickel.rs

```diff
@@ -178,18 +179,17 @@ impl<D: Sync + Send + 'static> Nickel<D> {
 
     /// Bind and listen for connections on the given host and port.
     ///
-    /// # Panics
-    ///
-    /// Panics if `addr` is an invalid `SocketAddr`.
-    ///
     /// # Examples
-    /// ```{rust,no_run}
+    /// ```rust
     /// use nickel::Nickel;
     ///
     /// let server = Nickel::new();
-    /// server.listen("127.0.0.1:6767");
+    /// let listening = server.listen("127.0.0.1:6767").expect("Failed to launch server");
+    /// println!("Listening on: {:?}", listening.socket());
```

Comment:
It was to show the use of `socket()` to get the address, but I guess this example will print twice (since the options were not passed to disable printing). So I think it's fine, but I should make the intention more explicit perhaps?